data/random-teams.ts

```diff
@@ -1151,8 +1150,9 @@ export class RandomTeams {
 		role: string,
 	): string | undefined {
 		if (
-			counter.get('Physical') >= 4 &&
-			['fakeout', 'firstimpression', 'flamecharge', 'rapidspin', 'ruination', 'superfang'].every(m => !moves.has(m))
+			(counter.get('Physical') >= 4 &&
+			['fakeout', 'firstimpression', 'flamecharge', 'rapidspin', 'ruination', 'superfang'].every(m => !moves.has(m))) ||
+			(counter.get('Physical') >= 3 && moves.has('memento'))
```

Comment:
```suggestion
			(counter.get('Physical') >= 4 ||
			(counter.get('Physical') >= 3 && moves.has('memento'))) &&
			['fakeout', 'firstimpression', 'flamecharge', 'rapidspin', 'ruination', 'superfang'].every(m => !moves.has(m))
```
Thankfully it doesn't matter for any set that currently exists, but your conditions are in the wrong order, unless you intended to allow band with something like Memento and Fake Out.